tinymod/plugins/code_metrics.py

```diff
@@ -0,0 +1,209 @@
+import math
+import shelve
+from typing import Annotated, Optional
+import hashlib
+import tempfile
+from asyncio.subprocess import PIPE
+import os
+from datetime import datetime, timedelta
+import tokenize
+import token
+import gzip
+from io import BytesIO
+import pygal
+from pygal.style import NeonStyle
+from PIL import Image
+import asyncio 
+import glob
+import numpy as np
+from hata import Client, Guild, Role
+from hata.ext.slash import InteractionResponse
+
+TinyMod: Client
+GUILD: Guild
+ADMIN_ROLE: Role
+
+REPO = "https://github.com/tinygrad/tinygrad.git"
+BRANCH = "master"
+FILE_FILTER = "tinygrad/**/*.py"
+WORKING_DIR = tempfile.gettempdir() + "/" + hashlib.sha256(REPO.encode("utf-8")).hexdigest()
+CACHE = shelve.open(WORKING_DIR + ".cache")
```

Comment:
Maybe cache the old metrics in a different directory? Because `/tmp` is commonly wiped on reboot. The repo is fine in `/tmp` tho.

Reply:
yeah makes sense. Did this for dev purposes. Will move it to `./.metrics.cache`